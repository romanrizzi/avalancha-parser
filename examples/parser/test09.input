
-- Disyunci√≥n

check A or B 
check A or B or C
check A or B or C or D
check not A or not B or not C or not D

check A1==A2 or B1==B2
check A1==A2 or B1==B2 or C1==C2
check A1==A2 or B1==B2 or C1==C2 or D1==D2
check not A1==A2 or not B1==B2 or not C1==C2 or not D1==D2

check A or B 
check (A or B) or C
check A or (B or C) or D
check not ((A or not (B or not C)) or not D)

check A1==A2 or B1==B2
check A1==A2 or (B1==B2 or C1==C2)
check ((A1==A2 or B1==B2) or C1==C2) or D1==D2
check not (A1==A2 or not (B1==B2 or not (C1==C2 or not (D1==D2))))

