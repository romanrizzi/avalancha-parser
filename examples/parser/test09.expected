[
   "program",
   [],
   [
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "or",
               [
                  "equal",
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "or",
               [
                  "equal",
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ],
               [
                  "or",
                  [
                     "equal",
                     [
                        "cons",
                        "C",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "D",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "not",
               [
                  "equal",
                  [
                     "cons",
                     "A",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ],
            [
               "or",
               [
                  "not",
                  [
                     "equal",
                     [
                        "cons",
                        "B",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ],
               [
                  "or",
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "C",
                           []
                        ],
                        [
                           "cons",
                           "True",
                           []
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "D",
                           []
                        ],
                        [
                           "cons",
                           "True",
                           []
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B1",
                  []
               ],
               [
                  "cons",
                  "B2",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "or",
               [
                  "equal",
                  [
                     "cons",
                     "B1",
                     []
                  ],
                  [
                     "cons",
                     "B2",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C1",
                     []
                  ],
                  [
                     "cons",
                     "C2",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "or",
               [
                  "equal",
                  [
                     "cons",
                     "B1",
                     []
                  ],
                  [
                     "cons",
                     "B2",
                     []
                  ]
               ],
               [
                  "or",
                  [
                     "equal",
                     [
                        "cons",
                        "C1",
                        []
                     ],
                     [
                        "cons",
                        "C2",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "D1",
                        []
                     ],
                     [
                        "cons",
                        "D2",
                        []
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "not",
               [
                  "equal",
                  [
                     "cons",
                     "A1",
                     []
                  ],
                  [
                     "cons",
                     "A2",
                     []
                  ]
               ]
            ],
            [
               "or",
               [
                  "not",
                  [
                     "equal",
                     [
                        "cons",
                        "B1",
                        []
                     ],
                     [
                        "cons",
                        "B2",
                        []
                     ]
                  ]
               ],
               [
                  "or",
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "C1",
                           []
                        ],
                        [
                           "cons",
                           "C2",
                           []
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "D1",
                           []
                        ],
                        [
                           "cons",
                           "D2",
                           []
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "or",
               [
                  "equal",
                  [
                     "cons",
                     "A",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "C",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "or",
               [
                  "or",
                  [
                     "equal",
                     [
                        "cons",
                        "B",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "C",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "D",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "not",
            [
               "or",
               [
                  "or",
                  [
                     "equal",
                     [
                        "cons",
                        "A",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "not",
                     [
                        "or",
                        [
                           "equal",
                           [
                              "cons",
                              "B",
                              []
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ],
                        [
                           "not",
                           [
                              "equal",
                              [
                                 "cons",
                                 "C",
                                 []
                              ],
                              [
                                 "cons",
                                 "True",
                                 []
                              ]
                           ]
                        ]
                     ]
                  ]
               ],
               [
                  "not",
                  [
                     "equal",
                     [
                        "cons",
                        "D",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B1",
                  []
               ],
               [
                  "cons",
                  "B2",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "or",
               [
                  "equal",
                  [
                     "cons",
                     "B1",
                     []
                  ],
                  [
                     "cons",
                     "B2",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C1",
                     []
                  ],
                  [
                     "cons",
                     "C2",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "or",
            [
               "or",
               [
                  "or",
                  [
                     "equal",
                     [
                        "cons",
                        "A1",
                        []
                     ],
                     [
                        "cons",
                        "A2",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "B1",
                        []
                     ],
                     [
                        "cons",
                        "B2",
                        []
                     ]
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C1",
                     []
                  ],
                  [
                     "cons",
                     "C2",
                     []
                  ]
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "D1",
                  []
               ],
               [
                  "cons",
                  "D2",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "not",
            [
               "or",
               [
                  "equal",
                  [
                     "cons",
                     "A1",
                     []
                  ],
                  [
                     "cons",
                     "A2",
                     []
                  ]
               ],
               [
                  "not",
                  [
                     "or",
                     [
                        "equal",
                        [
                           "cons",
                           "B1",
                           []
                        ],
                        [
                           "cons",
                           "B2",
                           []
                        ]
                     ],
                     [
                        "not",
                        [
                           "or",
                           [
                              "equal",
                              [
                                 "cons",
                                 "C1",
                                 []
                              ],
                              [
                                 "cons",
                                 "C2",
                                 []
                              ]
                           ],
                           [
                              "not",
                              [
                                 "equal",
                                 [
                                    "cons",
                                    "D1",
                                    []
                                 ],
                                 [
                                    "cons",
                                    "D2",
                                    []
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ]
   ]
]
